### PR TITLE
Add "show service sdp-using" parser for Nokia SROS

### PR DESCRIPTION
**Pull request by mickopp**
## Description
Adding parser for "show service sdp-using" cmd on Nokia SROS plattform

## Motivation and Context
Working on a multi vendor solution and starting verification tests

## Impact (If any)
No negative impact expected

## Screenshots:
python3 folder_parsing_job.py -o sros -c ShowServiceSdpUsing
2022-03-07T11:46:24: %AETEST-INFO: +------------------------------------------------------------------------------+
2022-03-07T11:46:24: %AETEST-INFO: |                   Starting testcase SuperFileBasedTesting                    |
2022-03-07T11:46:24: %AETEST-INFO: +------------------------------------------------------------------------------+
2022-03-07T11:46:24: %AETEST-INFO: +------------------------------------------------------------------------------+
2022-03-07T11:46:24: %AETEST-INFO: |                            Starting section setup                            |
2022-03-07T11:46:24: %AETEST-INFO: +------------------------------------------------------------------------------+
2022-03-07T11:46:24: %AETEST-INFO: The result of section setup is => PASSED
2022-03-07T11:46:24: %AETEST-INFO: The result of testcase SuperFileBasedTesting is => PASSED
2022-03-07T11:46:24: %AETEST-INFO: +------------------------------------------------------------------------------+
2022-03-07T11:46:24: %AETEST-INFO: |                            Starting testcase sros                            |
2022-03-07T11:46:24: %AETEST-INFO: +------------------------------------------------------------------------------+
2022-03-07T11:46:24: %AETEST-INFO: +------------------------------------------------------------------------------+
2022-03-07T11:46:24: %AETEST-INFO: |                            Starting section setup                            |
2022-03-07T11:46:24: %AETEST-INFO: +------------------------------------------------------------------------------+
2022-03-07T11:46:24: %AETEST-INFO: The result of section setup is => PASSED
2022-03-07T11:46:24: %AETEST-INFO: +------------------------------------------------------------------------------+
2022-03-07T11:46:24: %AETEST-INFO: |                     Starting section ShowServiceSdpUsing                     |
2022-03-07T11:46:24: %AETEST-INFO: +------------------------------------------------------------------------------+
2022-03-07T11:46:24: %AETEST-INFO: +..............................................................................+
2022-03-07T11:46:24: %AETEST-INFO: :                 Starting STEP 1: sros -> ShowServiceSdpUsing                 :
2022-03-07T11:46:24: %AETEST-INFO: +..............................................................................+
2022-03-07T11:46:24: %AETEST-INFO: +..............................................................................+
2022-03-07T11:46:24: %AETEST-INFO: :        Starting STEP 1.1: Test Golden -> sros -> ShowServiceSdpUsing         :
2022-03-07T11:46:24: %AETEST-INFO: +..............................................................................+
2022-03-07T11:46:24: %AETEST-INFO: +..............................................................................+
2022-03-07T11:46:24: %AETEST-INFO: :    Starting STEP 1.1.1: Gold -> sros -> ShowServiceSdpUsing -> golden_out    :
2022-03-07T11:46:24: %AETEST-INFO: :                                     put1                                     :
2022-03-07T11:46:24: %AETEST-INFO: +..............................................................................+
2022-03-07T11:46:24: %AETEST-INFO: The result of STEP 1.1.1: Gold -> sros -> ShowServiceSdpUsing -> golden_output1 is => PASSED
2022-03-07T11:46:24: %AETEST-INFO: The result of STEP 1.1: Test Golden -> sros -> ShowServiceSdpUsing is => PASSED
2022-03-07T11:46:24: %AETEST-INFO: +..............................................................................+
2022-03-07T11:46:24: %AETEST-INFO: :         Starting STEP 1.2: Test Empty -> sros -> ShowServiceSdpUsing         :
2022-03-07T11:46:24: %AETEST-INFO: +..............................................................................+
2022-03-07T11:46:24: %AETEST-INFO: +..............................................................................+
2022-03-07T11:46:24: %AETEST-INFO: :      Starting STEP 1.2.1: Empty -> sros -> ShowServiceSdpUsing -> empty      :
2022-03-07T11:46:24: %AETEST-INFO: +..............................................................................+
2022-03-07T11:46:24: %AETEST-INFO: The result of STEP 1.2.1: Empty -> sros -> ShowServiceSdpUsing -> empty is => PASSED
2022-03-07T11:46:24: %AETEST-INFO: The result of STEP 1.2: Test Empty -> sros -> ShowServiceSdpUsing is => PASSED
2022-03-07T11:46:24: %AETEST-INFO: The result of STEP 1: sros -> ShowServiceSdpUsing is => PASSED
2022-03-07T11:46:24: %AETEST-INFO: +..........................................................+
2022-03-07T11:46:24: %AETEST-INFO: :                       STEPS Report                       :
2022-03-07T11:46:24: %AETEST-INFO: +..........................................................+
2022-03-07T11:46:24: %AETEST-INFO: STEP 1 - sros -> ShowServiceSdpUsing              Passed
2022-03-07T11:46:24: %AETEST-INFO: STEP 1.1 - Test Golden -> sros -> ShowServiceSdpUsingPassed
2022-03-07T11:46:24: %AETEST-INFO: STEP 1.1.1 - Gold -> sros -> ShowServiceSdpUsing -> golden_output1Passed
2022-03-07T11:46:24: %AETEST-INFO: STEP 1.2 - Test Empty -> sros -> ShowServiceSdpUsingPassed
2022-03-07T11:46:24: %AETEST-INFO: STEP 1.2.1 - Empty -> sros -> ShowServiceSdpUsing -> emptyPassed
2022-03-07T11:46:24: %AETEST-INFO: ............................................................
2022-03-07T11:46:24: %AETEST-INFO: The result of section ShowServiceSdpUsing is => PASSED
2022-03-07T11:46:24: %AETEST-INFO: +------------------------------------------------------------------------------+
2022-03-07T11:46:24: %AETEST-INFO: |                           Starting section cleanup                           |
2022-03-07T11:46:24: %AETEST-INFO: +------------------------------------------------------------------------------+
2022-03-07T11:46:24: %AETEST-INFO: The result of section cleanup is => PASSED
2022-03-07T11:46:24: %AETEST-INFO: The result of testcase sros is => PASSED
2022-03-07T11:46:24: %GENIE-INFO: +------------------------------------------------------------------------------+
2022-03-07T11:46:24: %GENIE-INFO: |                               Unittest results                               |
2022-03-07T11:46:24: %GENIE-INFO: +------------------------------------------------------------------------------+
2022-03-07T11:46:24: %GENIE-INFO:  SECTIONS/TESTCASES                                                      RESULT
2022-03-07T11:46:24: %GENIE-INFO: --------------------------------------------------------------------------------
2022-03-07T11:46:24: %GENIE-INFO: .
2022-03-07T11:46:24: %GENIE-INFO: |-- SuperFileBasedTesting                                                 PASSED
2022-03-07T11:46:24: %GENIE-INFO: |   `-- setup                                                             PASSED
2022-03-07T11:46:24: %GENIE-INFO: `-- sros                                                                  PASSED
2022-03-07T11:46:24: %GENIE-INFO:     |-- setup                                                             PASSED
2022-03-07T11:46:24: %GENIE-INFO:     |-- ShowServiceSdpUsing                                               PASSED
2022-03-07T11:46:24: %GENIE-INFO:     |   |-- Step 1: sros -> ShowServiceSdpUsing                           PASSED
2022-03-07T11:46:24: %GENIE-INFO:     |   |-- Step 1.1: Test Golden -> sros -> ShowServiceSdpUsing          PASSED
2022-03-07T11:46:24: %GENIE-INFO:     |   |-- Step 1.1.1: Gold -> sros -> ShowServiceSdpUsing -> gold...    PASSED
2022-03-07T11:46:24: %GENIE-INFO:     |   |-- Step 1.2: Test Empty -> sros -> ShowServiceSdpUsing           PASSED
2022-03-07T11:46:24: %GENIE-INFO:     |   `-- Step 1.2.1: Empty -> sros -> ShowServiceSdpUsing -> empty     PASSED
2022-03-07T11:46:24: %GENIE-INFO:     `-- cleanup                                                           PASSED
2022-03-07T11:46:24: %GENIE-INFO: +------------------------------------------------------------------------------+
2022-03-07T11:46:24: %GENIE-INFO: |                                   Summary                                    |
2022-03-07T11:46:24: %GENIE-INFO: +------------------------------------------------------------------------------+
2022-03-07T11:46:24: %GENIE-INFO:  Number of ABORTED                                                            0
2022-03-07T11:46:24: %GENIE-INFO:  Number of BLOCKED                                                            0
2022-03-07T11:46:24: %GENIE-INFO:  Number of ERRORED                                                            0
2022-03-07T11:46:24: %GENIE-INFO:  Number of FAILED                                                             0
2022-03-07T11:46:24: %GENIE-INFO:  Number of PASSED                                                             2
2022-03-07T11:46:24: %GENIE-INFO:  Number of PASSX                                                              0
2022-03-07T11:46:24: %GENIE-INFO:  Number of SKIPPED                                                            0
2022-03-07T11:46:24: %GENIE-INFO:  Total Number                                                                 2
2022-03-07T11:46:24: %GENIE-INFO:  Success Rate                                                            100.0%
2022-03-07T11:46:24: %GENIE-INFO: --------------------------------------------------------------------------------
2022-03-07T11:46:24: %GENIE-INFO:  Total Passing Unittests                                                      2
2022-03-07T11:46:24: %GENIE-INFO:  Total Failed Unittests                                                       0
2022-03-07T11:46:24: %GENIE-INFO:  Total Errored Unittests                                                      0
2022-03-07T11:46:24: %GENIE-INFO:  Total Unittests                                                              2
2022-03-07T11:46:24: %GENIE-INFO: --------------------------------------------------------------------------------

## Checklist:
- [x] I have updated the changelog.
- [ ] I have updated the documentation (If applicable).
- [x] I have added tests to cover my changes (If applicable).
- [x] All new and existing tests passed.
- [x] All new code passed compilation.

### CODE REVIEW

File: src/genie/libs/parser/sros/show_service_sdp_using.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+class ShowServiceSdpUsing(ShowServiceSdpUsingSchema):
+    """ Parser for show service sdp-using"""
+    cli_command = 'show service sdp-using'
+
+    def cli(self, output=None):
+
+        if output is None:
+            out = self.device.execute(self.cli_command)
+        else:
+            out = output
```
Comment (danielgraziano) [inline src/genie/libs/parser/sros/show_service_sdp_using.py:36]:
```suggestion
        if output is None:
            output = self.device.execute(self.cli_command)
```

then on line 57 you can do
``` python
for line in output.splitlines():

File: src/genie/libs/parser/sros/show_service_sdp_using.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+                line = line.strip()
+            else:
+                continue
+
+            #===============================================================================
+            #SDP Using
+            #===============================================================================
+            #SvcId      SdpId              Type   Far End              Opr   I.Label E.Label
+            #                                                          State
+            #-------------------------------------------------------------------------------
```
Comment (danielgraziano) [inline src/genie/libs/parser/sros/show_service_sdp_using.py:68]: please remove

**Review by danielgraziano — CHANGES_REQUESTED** [review]
Looks good after these changes

File: src/genie/libs/parser/sros/show_service_sdp_using.py (hunk 1 above)
Comment (mickopp) [reply src/genie/libs/parser/sros/show_service_sdp_using.py]: Hi Daniel,
thansk for the review - I commited the suggested changes and found during testing an additional cornercase, therefor I had to fix the parser (change label to 'str' as in 'None' label ist a valid output) and extended the golden output testfiles
thanks and regards
Michael

**Review by danielgraziano — APPROVED** [review]

**Review by lsheikal — CHANGES_REQUESTED** [review]
Please fix the merge conflict.

**Comment by mickopp** [comment]
Hi all, just a quick question for further pull requests.
Should the "github_parser.json" file generated by "make json" be commited in a pull request or not ? What is the current best practice ?
As this file cause merge conflicts due to other merges

**Comment by lsheikal** [comment]
> Hi all, just a quick question for further pull requests. Should the "github_parser.json" file generated by "make json" be commited in a pull request or not ? What is the current best practice ? As this file cause merge conflicts due to other merges

yes you have to commit the json file.

**Review by lsheikal — APPROVED** [review]

**Pull request merged**